crates/bls-gadgets/src/y_to_bit.rs

```diff
@@ -22,8 +28,7 @@ impl<P: Bls12Parameters> YToBitGadget<P> {
         mut cs: CS,
         pk: &G1Gadget<P>,
     ) -> Result<Boolean, SynthesisError> {
-        let half_plus_one_neg =
-            (P::Fp::from_repr(P::Fp::modulus_minus_one_div_two()) + &P::Fp::one()).neg();
+        let half_neg = (P::Fp::from_repr(P::Fp::modulus_minus_one_div_two())).neg();
```

Comment:
```suggestion
        let half_neg = P::Fp::from_repr(P::Fp::modulus_minus_one_div_two()).neg();
```

The extra parenthesis shouldn't be needed?